include/aws/io/channel.h

```diff
@@ -59,6 +60,9 @@ struct aws_channel {
     struct aws_shutdown_notification_task shutdown_notify_task;
     aws_channel_on_shutdown_completed_fn *on_shutdown_completed;
     void *shutdown_user_data;
+    struct aws_linked_list pending_tasks;
+    struct aws_mutex x_thread_task_lock;
```

Comment:
So near as I can tell, this mutex protects only this cross thread task queue, is that correct? Probably should note that here. Maybe even shove them in a struct together to make it super clear.